src/python/espressomd/visualization.pyx

```diff
@@ -0,0 +1,209 @@
+        else:
+            self.points.mlab_source.reset(x=coords[:,0]%boxl[0], y=coords[:,1]%boxl[1], z=coords[:,2]%boxl[2], u=radii, v=radii, w=radii, scalars=types)
+        if not self.running:
+            f.scene.reset_zoom()
+            self.running = True
+
+        if not Nbonds_changed:
+            if bonds.shape[0] > 0:
+                self.arrows.mlab_source.set  (x=bonds[:,0], y=bonds[:,1], z=bonds[:,2], u=bonds[:,3], v=bonds[:,4], w=bonds[:,5])
+        else:
+            self.arrows.mlab_source.reset(x=bonds[:,0], y=bonds[:,1], z=bonds[:,2], u=bonds[:,3], v=bonds[:,4], w=bonds[:,5], scalars=bonds[:,6])
+
+    def update(self):
+        """Pull the latest particle information from Espresso.
+        This is the only function that should be called from the computation thread.
+        It does not call any Mayavi functions unless it is being called from the main (GUI) thread."""
+
+        if self.last_T is not None and self.last_T == self.system.time:
+            return
+        self.last_T = self.system.time
+        
+        cdef int N = self.system.n_part
+        coords = numpy.zeros((N,3))
+        types = numpy.empty(N, dtype=int)
+        inter = NonBondedInteractions()
+        radii = numpy.empty(N)
+        cdef int i=0,j=0,k=0 
+        cdef int t 
+        cdef int partner
+        cdef particle p
+        cdef ia_parameters* ia
+        cdef vector[int] bonds
+
+    # Using (additional) untyped variables and python constructs in the loop 
+    # will slow it down considerably. 
+        for i in range(self.system.max_part+1):
+            if not particle_exists(i):
+                continue
+            get_particle_data(i,&p)
+            coords[j,:] = p.r.p
+            t = p.p.type
+            types[j] = t +1
+            radii[j] =get_ia_param(t,t).LJ_sig *0.5
+
+            # ITerate over bonds
+            k=0        
+            while k<p.bl.n:
+                # Bond type
+                t= p.bl.e[k]
+                k+=1
+                # Iterate over bond partners and store each connection
+                for l in range(bonded_ia_params[t].num):
+                    bonds.push_back(i)
+                    bonds.push_back(p.bl.e[k])
+                    k+=1
+            j += 1
+        assert j == self.system.n_part
+        cdef int Nbonds = bonds.size()/2
+        
+        bond_coords = numpy.zeros((Nbonds,7))
+
+        cdef int n
+        cdef particle p1,p2
+        cdef double bond_vec[3]
+        for n in range(Nbonds):
+            i =bonds[2*n]
+            j =bonds[2*n+1]
+            get_particle_data(i,&p1)
+            get_particle_data(j,&p2)
+            bond_coords[n,:3] = p1.r.p 
+            get_mi_vector(bond_vec,p2.r.p,p1.r.p)
+            bond_coords[n,3:6] = bond_vec
+# Not needed            bond_coords[n,6] = 0 # numeric bond IDs have been removed
```

Comment:
If you store the bond type into the bonds array in line 148, you could use it here to color the bonds based on type:
```
i =bonds[3*n]
j =bonds[3*n+1]
t =bonds[3*n+2]
[...]
bond_coords[n,6] = t
```